core/src/test/scala/akka/kafka/internal/ProducerTest.scala

```diff
@@ -90,6 +91,22 @@ class ProducerTest(_system: ActorSystem)
       client.verifyNoMoreInteractions()
     }
   }
+  it should "work with a provided Producer" in {
+    assertAllStagesStopped {
+      val input = 1 to 10 map { recordAndMetadata(_)._1 }
+
+      val mockProducer = new MockProducer[String, String]()
+
+      val fut: Future[Done] = Source(input).runWith(Producer.plainSink(settings, mockProducer))
+
+      Thread.sleep(1000)
+      mockProducer.flush()
```

Comment:
found I have to call flush (after sleeping) to get the stream to actually close. Perhaps if I push a larger set of data it will work....